### PR TITLE
[clang][Sema] Resolving Inconsistent Arguments Panic in Variadic Template Variables

### PR DESCRIPTION
**Pull request by knightXun**
When template variables are variadic, sema may panic, potentially leading to a situation where the number of function variables exceeds the number of template variables.The sema compares the template signature and function signature parameters one by one, which can trigger an assertion error. This PR, when encountering variadic templates,
avoids comparing the template signature and function signature parameters one by one.

[#70191](https://github.com/llvm/llvm-project/issues/70191)

### CODE REVIEW

**Comment by knightXun** [comment]
i will add test case for it

**Comment by cor3ntin** [comment]
@knightXun code looks reasonable, ping me when you have tests and a release note :)

**Comment by knightXun** [comment]
> @knightXun code looks reasonable, ping me when you have tests and a release note :)
got it,

**Comment by knightXun** [comment]
cc @cor3ntin

**Review by shafik — COMMENTED** [review]
@cor3ntin do this still look good to you?

File: clang/test/SemaCXX/GH70280.cpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// RUN: %clang_cc1 -fsyntax-only -verify %s
+
+namespace PR70280 {
+  typedef a; // expected-error {{a type specifier is required for all declarations}}
+  using b = char*;
+  template <typename... c> void d(c...) = d<b, a>(0, ""); // expected-error {{no matching function for call to 'd'}}
+  // expected-error@-1 {{illegal initializer (only variables can be initialized)}}
+  // expected-note@-2 {{candidate function template not viable: no known conversion from 'const char[1]' to 'a' (aka 'int') for 2nd argument}}
+}
```
Comment (cor3ntin) [inline clang/test/SemaCXX/GH70280.cpp:9]: Missing EOL

**Review by cor3ntin — APPROVED** [review]

**Comment by cor3ntin** [comment]
@shafik Yes. It needs to be rebased though
@knightXun I apologize for the long delay. In the future, feel free to ping if you don't get a reply after one or 2 weeks

**Comment by cor3ntin** [comment]
Fixed by https://github.com/llvm/llvm-project/commit/7a28a5b3fee6c78ad59af79a3d03c00db153c49f

**Pull request closed without merging**